src/backend/crossterm.rs

```diff
@@ -36,149 +155,96 @@ impl Backend {
+                        position = (x, y).into();
                     }
-
-                    Event::Mouse {
-                        event,
-                        position,
-                        offset: Vec2::zero(),
+                    CMouseEvent::Up(_, x, y, _) => {
+                        event = MouseEvent::Release(self.last_button.unwrap());
+                        position = (x, y).into();
                     }
-                }
-                CMouseEvent::Release(x, y) if self.last_button.is_some() => {
-                    let event = MouseEvent::Release(self.last_button.unwrap());
-                    let position = (x, y).into();
-
-                    Event::Mouse {
-                        event,
-                        position,
-                        offset: Vec2::zero(),
+                    CMouseEvent::Drag(_, x, y, _) => {
+                        event = MouseEvent::Hold(self.last_button.unwrap());
+                        position = (x, y).into();
                     }
-                }
-                CMouseEvent::Hold(x, y) if self.last_button.is_some() => {
-                    let event = MouseEvent::Hold(self.last_button.unwrap());
-                    let position = (x, y).into();
-
-                    Event::Mouse {
-                        event,
-                        position,
-                        offset: Vec2::zero(),
+                    CMouseEvent::ScrollDown(x, y, _) => {
+                        event = MouseEvent::WheelDown;
+                        position = (x, y).into();
                     }
+                    CMouseEvent::ScrollUp(x, y, _) => {
+                        event = MouseEvent::WheelDown;
+                        position = (x, y).into();
+                    }
+                };
+
+                Event::Mouse {
+                    event,
+                    position,
+                    offset: Vec2::zero(),
                 }
-                _ => {
-                    log::warn!(
-                        "Unknown mouse button event {:?}!",
-                        mouse_event
-                    );
-                    Event::Unknown(vec![])
-                }
-            },
-            _ => {
-                log::warn!("Unknown mouse event {:?}!", event);
-                Event::Unknown(vec![])
             }
+            CEvent::Resize(_, _) => Event::WindowResize,
         }
     }
 }
 
 impl backend::Backend for Backend {
-    fn name(&self) -> &str {
-        "crossterm"
-    }
-
     fn poll_event(&mut self) -> Option<Event> {
-        self.async_reader.next().map(|event| self.map_key(event))
+        match poll(Duration::from_millis(5)) {
+            Ok(true) => match read() {
+                Ok(event) => Some(self.map_key(event)),
+                Err(_) => panic!(),
+            },
+            _ => None,
+        }
     }
 
     fn finish(&mut self) {
-        input().disable_mouse_mode().unwrap();
-        cursor().show().unwrap();
```

Comment:
please don't forget to re-show the cursor !
(tried on Ubuntu)

Reply:
Didn't it work on ubuntu? It is supposed to so with this [statement](https://github.com/TimonPost/cursive/blob/crossterm_update/src/backend/crossterm.rs#L242).

Reply:
exactly; with current master, with crossterm backend, the cursor is visible after exiting each of the cursive demos, with this branch, it isn't
i've tested in gnome-terminal and inside tmux, same effect!

Reply:
I fixed the issue

Reply:
yes this works now thank youu